test-adapter/src/main/scala/org/scalajs/sbttestadapter/ScalaJSRunner.scala

```diff
@@ -61,28 +63,37 @@ final class ScalaJSRunner private[testadapter] (
     taskInfos.map(ScalaJSTask.fromInfo(this, _)).toArray
   }
 
-  def done(): String = synchronized {
-    ensureNotDone()
-
-    stopSlaves()
-
-    master.send("runnerDone")
-    val summary = ComUtils.receiveResponse(master) {
-      case ("ok", summary) => summary
+  def done(): String = {
+    synchronized {
+      ensureNotDone()
     }
```

Comment:
You cannot split the synchronized block as this can expose race conditions to the framework user. (If two threads call done, one of them must throw).